adobe_vipm/flows/validation/transfer.py

```diff
@@ -304,205 +314,195 @@ def validate_transfer_not_migrated(mpt_client, adobe_client, order):
     )
 
 
-def validate_transfer(mpt_client, adobe_client, order):
-    """
-    Validates a transfer order.
-
-    Args:
-        mpt_client (MPTClient): The client used to consume the MPT API.
-        adobe_client (AdobeClient): The client used to consume the Adobe VIPM API.
-        order (dict): The order to validate.
-
-    Returns:
-        tuple: (True, order) if there is a validation error, (False, order) otherwise.
-    """
-    config = get_config()
-    authorization_id = order["authorization"]["id"]
-    authorization = config.get_authorization(authorization_id)
-    membership_id = get_adobe_membership_id(order)
-    product_id = order["agreement"]["product"]["id"]
-
-    transfer = get_transfer_by_authorization_membership_or_customer(
-        product_id,
-        authorization.authorization_id,
-        membership_id,
-    )
-
-    if not transfer:
-        return validate_transfer_not_migrated(mpt_client, adobe_client, order)
-
-    has_error, order = _validate_transfer_status(transfer, order)
-    if has_error:
-        return True, order
-
-    try:
-        subscriptions, adobe_transfer = _get_transfer_data(
-            adobe_client,
-            authorization_id,
-            transfer
+class SetupTransferContext(Step):
+    def __call__(self, mpt_client, context, next_step):
+        context.validation_succeeded = True
+        config = get_config()
+        context.order = reset_ordering_parameters_error(context.order)
+        context.order = reset_order_error(context.order)
+        context.order["agreement"] = get_agreement(mpt_client, context.order["agreement"]["id"])
+        context.order["agreement"]["licensee"] = get_licensee(
+            mpt_client, context.order["agreement"]["licensee"]["id"]
         )
-    except AdobeError as e:
-        order = set_ordering_parameter_error(
+        product_id = context.order["agreement"]["product"]["id"]
+        authorization_id = context.order["authorization"]["id"]
+        authorization = config.get_authorization(authorization_id)
+        membership_id = get_adobe_membership_id(context.order)
+        transfer = get_transfer_by_authorization_membership_or_customer(
+            product_id,
+            authorization.authorization_id,
+            membership_id,
+        )
+        context.transfer = transfer
+        context.authorization = authorization
+        context.membership_id = membership_id
+
+        next_step(mpt_client, context)
+
+class ValidateTransferStatus(Step):
+    def __call__(self, mpt_client, context, next_step):
+        transfer = context.transfer
+        order = context.order
+        if not transfer:
+            next_step(mpt_client, context)
```

Comment:
For what reason you need to proceed with validation if transfer does not exists?